### PR TITLE
(web) show project id for cli users

### PR DESCRIPTION
**Pull request by joewagner**
fixes https://linear.app/tableland/issue/ENG-831/studio-web-app-should-show-me-project-id-somewhere

If you are trying to use the CLI in combination with the web app, it will be helpful to have a way to find and copy a project id.
This PR enables that, and does a small refactor of the HashDisplay component to allow showing the entire hash.

### CODE REVIEW

**Comment by joewagner** [comment]
I'm moving this back to draft.  I'm gonna look at updating the layout of the descriptions, and I found an issue where webpack doesn't like a div being a descendant of a p.

**Comment by joewagner** [comment]
I added an attempt to make the project page layout a bit more _efficient_.  
Here are screengrabs for two different projects.  One with a long description, and one with a short description.
There's an image for each project at screen widths of 1200px, 800px, and 600px.

1200px
Screen Shot 2024-03-20 at 10 22 33 AM
Screen Shot 2024-03-20 at 10 22 44 AM

800px
Screen Shot 2024-03-20 at 10 23 25 AM
Screen Shot 2024-03-20 at 10 23 38 AM

600px
Screen Shot 2024-03-20 at 10 24 10 AM
Screen Shot 2024-03-20 at 10 24 25 AM

File: packages/web/tailwind.config.js [hunk 1]
```diff
@@ -15,6 +15,16 @@ module.exports = {
         "2xl": "1400px",
       },
     },
+    minWidth: ({ theme }) => ({
+      ...theme("spacing"),
+      full: "100%",
+      "1/4": "25%",
```
Comment (joewagner) [inline packages/web/tailwind.config.js:21]: tailwind doesn't include min width partial percentages by default ref: https://github.com/tailwindlabs/tailwindcss/blob/master/stubs/config.full.js#L685

File: packages/web/app/[team]/[project]/(project)/page.tsx [hunk 2]
```diff
@@ -38,14 +39,27 @@ export default async function Project({
...
+          <h2 className="text-lg font-semibold tracking-tight">
+            Project description
+          </h2>
+          <p className="ml-4 whitespace-pre-wrap leading-tight text-muted-foreground">
+            {project.description}
+          </p>
+        </div>
+        <div className="space-y-1 p-4">
+          <h2 className="text-lg font-semibold tracking-tight">Project ID</h2>
+          <span className="ml-4 block min-w-72 whitespace-pre-wrap leading-tight text-muted-foreground">
```
Comment (joewagner) [inline packages/web/app/[team]/[project]/(project)/page.tsx:54]: We don't want the address to ever be less than 18rem (288px) width, so I added the display: block and min width here

File: packages/web/app/[team]/[project]/(project)/page.tsx [hunk 3]
```diff
@@ -38,14 +39,27 @@ export default async function Project({
...
-      <div className="space-y-1">
-        <h2 className="text-lg font-semibold tracking-tight">
-          Project description
-        </h2>
-        <p className="ml-4 max-w-lg whitespace-pre-wrap leading-tight text-muted-foreground">
-          {project.description}
-        </p>
+    <main className="container flex flex-col space-y-6 p-4">
+      <div className="flex flex-col p-4 md:flex-row">
+        <div className="min-w-1/2 space-y-1 p-4">
```
Comment (joewagner) [inline packages/web/app/[team]/[project]/(project)/page.tsx:44]: In order to make sure that small descriptions take up at least half the width on large screens I added a min width of 50% here.

File: packages/web/app/[team]/[project]/(project)/page.tsx [hunk 4]
```diff
@@ -38,14 +39,27 @@ export default async function Project({
...
-    <main className="container flex flex-1 flex-col space-y-6 p-4">
-      <div className="space-y-1">
-        <h2 className="text-lg font-semibold tracking-tight">
-          Project description
-        </h2>
-        <p className="ml-4 max-w-lg whitespace-pre-wrap leading-tight text-muted-foreground">
-          {project.description}
-        </p>
+    <main className="container flex flex-col space-y-6 p-4">
+      <div className="flex flex-col p-4 md:flex-row">
```
Comment (joewagner) [inline packages/web/app/[team]/[project]/(project)/page.tsx:43]: For screens smaller than `md` we want to allow the layout to collapse into a single column, but for larger screens we want two columns.  This is achieved by using the `flex-col md:flex-row` classes together.

**Comment by dtbuchholz** [comment]
nice! one thing i just realized is we've had left margin on the project description and ID paragraphs. optionally, we could remove that so that it aligns with the header—i.e., it's currently set to `ml-4`, but i think it looks better w/o it:
Screenshot 2024-03-20 at 11 45 58 AM
...imo that's more natural, looks a bit weird to have it indented.

regardless, these improvements lgtm.

**Review by dtbuchholz — DISMISSED** [review]

**Comment by joewagner** [comment]
@dtbuchholz good call on the margins.  I might have messed them up when I was adding the responsive flexbox classes.
Pushed a fix, I'll need another 👍

**Comment by dtbuchholz** [comment]
@joewagner they were always there, actually—they're on the current site!

**Review by dtbuchholz — APPROVED** [review]

**Review by asutula — APPROVED** [review]

**Pull request merged**